src/useReactions.tsx

```diff
@@ -177,7 +191,53 @@ export const ReactionsProvider = ({
         return;
       }
 
-      if (event.getType() === EventType.Reaction) {
+      if (event.getType() === ElementCallReactionEventType) {
+        const content: ECallReactionEventContent = event.getContent();
+
+        const membershipEventId = content?.["m.relates_to"]?.event_id;
+        // Check to see if this reaction was made to a membership event (and the
+        // sender of the reaction matches the membership)
+        if (
+          !memberships.some(
+            (e) => e.eventId === membershipEventId && e.sender === sender,
+          )
+        ) {
+          logger.warn(
+            `Reaction target was not a membership event for ${sender}, ignoring`,
+          );
+          return;
+        }
+
+        const emoji = content.emoji?.split(/(?:)/u)?.[0];
```

Comment:
`Intl.Segmenter` would be more appropriate here, as not all emoji are a single codepoint

Reply:
Ah, this in theory does actually work although I'll take a more standard solution than my regex hack. I do remember that `Intl.Segmenter` caused us trouble with Firefox ESR in Element Web land, so I wonder if a polyfill may be appropriate here.